exes/mqtt-bridge/src/tfc_to_external.cpp

```diff
@@ -0,0 +1,238 @@
+
+    co_await std::visit(
+        [&signal_data, this]<typename recv_t>(recv_t&& receiver) -> asio::awaitable<void> {
+          using receiver_t = std::remove_cvref_t<decltype(receiver)>;
+          if constexpr (!std::same_as<std::monostate, receiver_t>) {
+            if (!signal_data.current_value.has_value()) {
+              co_await set_initial_value(receiver, signal_data);
+            }
+          }
+          co_return;
+        },
+        signal_data.receiver);
+
+    logger_.info("Signal_data processed and added to the payload: {}", signal_data.information.name);
+  }
+
+  spark_plug_interface_.set_current_values(signals_);
+  spark_plug_interface_.send_current_values();
+}
+
+template <class spark_plug_config_t, class mqtt_client_t, class ipc_client_t>
+auto tfc_to_external<spark_plug_config_t, mqtt_client_t, ipc_client_t>::async_receive_routine(auto&& receiver,
+                                                                                              bool& read_finished,
+                                                                                              structs::signal_data& signal)
+    -> asio::awaitable<void> {
+  auto val = co_await receiver->async_receive(asio::use_awaitable);
+  if (val.has_value()) {
+    signal.current_value = val.value();
+  }
+  read_finished = true;
+}
+
+// This function reads the initial value off the signal.
+// The ideal solution is to use awaitable operators, but they are not compatible with azmq for an unknown reason.
+template <class spark_plug_config_t, class mqtt_client_t, class ipc_client_t>
+auto tfc_to_external<spark_plug_config_t, mqtt_client_t, ipc_client_t>::set_initial_value(ipc::details::any_slot receiver,
+                                                                                          structs::signal_data& signal)
+    -> asio::awaitable<void> {
+  bool read_finished = false;
+
+  asio::steady_timer timer(co_await asio::this_coro::executor);
+  timer.expires_after(std::chrono::milliseconds(20));
+
+  timer.async_wait([this, &read_finished](std::error_code error_code) {
+    if (!error_code) {
+      cancel_signal_.emit(asio::cancellation_type::all);
+      read_finished = true;
+    }
+  });
+
+  std::visit(
+      [this, &read_finished, &signal]<typename recv_t>(recv_t&& recv) {
+        using receiver_t = std::remove_cvref_t<decltype(recv)>;
+        if constexpr (!std::same_as<std::monostate, receiver_t>) {
+          asio::co_spawn(spark_plug_interface_.strand(),
+                         async_receive_routine(std::forward<decltype(recv)>(recv), read_finished, signal),
+                         asio::bind_cancellation_slot(cancel_signal_.slot(), asio::detached));
+        }
+      },
+      receiver);
+
+  while (!read_finished) {
+    co_await asio::steady_timer{ spark_plug_interface_.strand(), std::chrono::milliseconds{ 50 } }.async_wait(
+        asio::use_awaitable);
+  }
+  co_return;
```

Comment:
Mjög vel athugað hjá þér með lifetimeið á timernum.
Auðvitað ef hann d'torar áður en tíminn er liðinn
þá er bara kallað á callback með villu og ekkert gerist.
Very töff.  Mátt samt bomba í 
```suggestion
  timer.cancel();
  co_return;
```
bara til að vera augljós.

Reply:
okei, geri það.